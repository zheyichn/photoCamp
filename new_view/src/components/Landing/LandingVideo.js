import React from "react";
import Box from '@mui/material/Box';
import CardMedia from '@mui/material/CardMedia';
import { Typography } from "@mui/material";
import { backgroundBoxStyles, videoHeaderStyles } from "../../styles/landingStyle";

export default function LandingVideo() {
  return (
    <Box sx={ backgroundBoxStyles } >
        <Box component="span" sx={ videoHeaderStyles } >
            <Typography>
                PhotoCamp Best Video 2022
            </Typography>
        </Box>
        <CardMedia
            component="video"
            height="250"
            width="500"
            image="https://upload.wikimedia.org/wikipedia/commons/f/ff/360_SKIING_MAY_2019.webm"
            title="Test Video"
            autoPlay
            muted
            loop
        />
    </Box>
  );
}