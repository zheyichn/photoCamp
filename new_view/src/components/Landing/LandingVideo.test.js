import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import LandingVideo from "./LandingVideo";

// afterEach function runs after each test suite is executed
afterEach(() => {
    cleanup();
});

describe("Landing Video", () => {
  
    // Test1: match snapshots
    it("Page renders correctly", () => {
      const tree = renderer
        .create(
          <LandingVideo />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
});