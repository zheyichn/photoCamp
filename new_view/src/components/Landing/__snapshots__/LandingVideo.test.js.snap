// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Video Page renders correctly 1`] = `
<div
  className="MuiBox-root css-jskp5d"
>
  <span
    className="MuiBox-root css-1a5zk3b"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      PhotoCamp Best Video 2022
    </p>
  </span>
  <video
    autoPlay={true}
    className="MuiCardMedia-root MuiCardMedia-media css-e6k82g-MuiCardMedia-root"
    height="250"
    loop={true}
    muted={true}
    src="https://upload.wikimedia.org/wikipedia/commons/f/ff/360_SKIING_MAY_2019.webm"
    title="Test Video"
    width="500"
  />
</div>
`;
